examples/example-ethers-v5/index.js

```diff
@@ -0,0 +1,307 @@
+
+  parentElement.innerHTML = getTokenInfoHTML(
+    selectedToken.name,
+    selectedToken.address,
+    selectedToken.network,
+    selectedToken.balance,
+    selectedToken.symbol
+  );
+  setAmountButton.removeAttribute('hidden');
+});
+
+// 6. Choose amount of token to be sent
+setAmountButton.addEventListener('click', async () => {
+  const input = parentElement.querySelector('input');
+  amount = Number(input.value);
+
+  if (!amount) {
+    alert('Choose amount.');
+    return;
+  }
+
+  if (amount > Number(selectedToken.balance)) {
+    alert('Insufficient Balance');
+    return;
+  }
+
+  if (!settings.external.isTopupEnabled) {
+    alert('Functionality is not available.');
+    return;
+  }
+
+  setAmountButton.setAttribute('hidden', '');
+  parentElement.innerHTML = getSpinnerHTML();
+
+  const response = await sdk.convertTokenToEUR(selectedToken.address, selectedToken.decimals, String(amount), selectedToken.network);
+
+  amountInEUR = response.EURAmount;
+
+  const isLessThanMinimum = Number(amountInEUR) < Number(settings.external.minTopUpAmountInEUR);
+  const isMoreThanMaximum = Number(amountInEUR) > Number(settings.external.maxTopUpAmountInEUR);
+  const isTestHolytag = holytag.toUpperCase() === 'SDKTEST';
+
+  if (!isTestHolytag && (isLessThanMinimum || isMoreThanMaximum)) {
+    parentElement.innerHTML = getTokenInfoHTML(
+      selectedToken.name,
+      selectedToken.address,
+      selectedToken.network,
+      selectedToken.balance,
+      selectedToken.symbol
+    );
+    setAmountButton.removeAttribute('hidden');
+  }
+
+  if (!isTestHolytag && isLessThanMinimum) {
+    alert(`Minimum allowed amount is ${settings.external.minTopUpAmountInEUR} EUR`);
+    return;
+  }
+
+  if (!isTestHolytag && isMoreThanMaximum) {
+    alert(`Maximum allowed amount is ${settings.external.minTopUpAmountInEUR} EUR`);
+    return;
+  }
+
+  transferData = response.transferData;
+
+  parentElement.innerHTML = getDataHTML(
+    selectedToken.name,
+    selectedToken.address,
+    selectedToken.network,
+    selectedToken.symbol,
+    amount,
+    amountInEUR,
+    holytag
+  );
+  submit.removeAttribute('hidden');
+});
+
+// 7. Submit sending of token to recipient's debit card (this could require more than one
+//    wallet interaction, e.g. sign permit and then send a transaction
+submitButton.addEventListener('click', async () => {
+  const { chainId } = await provider.getNetwork();
+  const tokenNetworkId = getNetworkChainId(selectedToken.network);
+
+  submitButton.setAttribute('hidden', '');
+  parentElement.innerHTML = `
+    ${getSpinnerHTML()}
+    <dl></dl>
+  `;
+
+  const dlElement = parentElement.querySelector('dl');
+
+  // switch to the correct chain (network) in the wallet
+  if (chainId !== tokenNetworkId) {
+    await provider.provider.request({
+      method: 'wallet_switchEthereumChain',
+      params: [{ chainId: ethers.utils.hexValue(tokenNetworkId) }]
+    })
+  }
```

Comment:
This condition appears to be missing 'missing chain' scenario, which will cause this provider request to fail if the wallet app does not know such network

Consider using more sophisticated error handling to trigger `wallet_addEthereumChain` in such cases